universal-application-tool-0.0.1/app/services/applicant/ApplicantServiceImpl.java

```diff
@@ -61,4 +99,59 @@ public ApplicantServiceImpl(
             },
             httpExecutionContext.current());
   }
+
+  /**
+   * Tries to perform an in-place update of {@link ApplicantData}. Returns error messages if it is
+   * impossible to perform the operation.
+   */
+  private ImmutableSet<String> stageUpdates(
+      Applicant applicant,
+      ProgramDefinition programDefinition,
+      long blockId,
+      ImmutableSet<Update> updates) {
+
+    Optional<BlockDefinition> blockDefinitionMaybe = programDefinition.getBlockDefinition(blockId);
+    if (blockDefinitionMaybe.isEmpty()) {
+      return ImmutableSet.of("Block not found.");
+    }
+    BlockDefinition blockDefinition = blockDefinitionMaybe.get();
+
+    ImmutableList<Path> updatePaths =
+        updates.stream().map(Update::path).collect(ImmutableList.toImmutableList());
+    if (!blockDefinition.hasPaths(updatePaths)) {
+      return ImmutableSet.of("Not all paths exist in this Block.");
+    }
```

Comment:
Issue (blocking): it looks like the paths are checked for presence in the block twice. Once here, and once in the other `stageUpdates` method.

Reply:
done